Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,847 @@
+            } else {
+                $SSLEnforcementRolledback = $_.SSLEnforcement
+                $tlsVersionRolledBack = $_.TLSVersion
+            }
+
+            $DBForPostgreSQLSS.TLSVersion = $tlsVersionRolledBack
+            $DBForPostgreSQLSS.prevTLSVersion = $TlsVersionBeforeRollback
+            $DBForPostgreSQLSS.SSLEnforcement = $SSLEnforcementRolledback
+            $DBForPostgreSQLSS.prevSSLEnforcement = $SSLEnforcementBeforeRollback
+            $DBForPostgreSQLSSRolledBack += $DBForPostgreSQLSS
+        }
+        catch {
+            $DBForPostgreSQLSSSkipped += $DBForPostgreSQLSS
+        }
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "Rollback Summary:`n" -ForegroundColor $([Constants]::MessageType.Info)
+
+    if ($($DBForPostgreSQLSSRolledBack | Measure-Object).Count -gt 0) {
+        Write-Host "TLS version and SSL enforcement has been rolled back on the following Azure Database for PostgreSQL single server(s) in the Subscription.:" -ForegroundColor $([Constants]::MessageType.Update)
+        $DBForPostgreSQLSSRolledBack | Format-Table -Property $colsProperty -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+
+        # Write this to a file.
+        $DBForPostgreSQLSSRolledBackFile = "$($backupFolderPath)\RolledBackDBForPostgreSQLSS.csv"
+        $DBForPostgreSQLSSRolledBack | Export-CSV -Path $DBForPostgreSQLSSRolledBackFile -NoTypeInformation
+        Write-Host "This information has been saved to" -NoNewline
+        Write-Host " [$($DBForPostgreSQLSSRolledBackFile)]" -ForegroundColor $([Constants]::MessageType.Update) 
+    }
+
+    if ($($DBForPostgreSQLSSSkipped | Measure-Object).Count -gt 0) {
+        Write-Host "Error while rolling back TLS version or SSL enforcement on Azure Database for PostgreSQL single server(s) in the Subscription.:" -ForegroundColor $([Constants]::MessageType.Error)
+        $DBForPostgreSQLSSSkipped | Format-Table -Property $colsProperty -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+
+        # Write this to a file.
+        $DBForPostgreSQLSSSkippedFile = "$($backupFolderPath)\RollbackSkippedDBForPostgreSQLSS.csv"
+        $DBForPostgreSQLSSSkipped | Export-CSV -Path $DBForPostgreSQLSSSkippedFile -NoTypeInformation
+        Write-Host "This information has been saved to" -NoNewline
+        Write-Host " [$($DBForPostgreSQLSSSkippedFile)]" -ForegroundColor $([Constants]::MessageType.Update) 
+    }
+    Write-Host $([Constants]::DoubleDashLine)
+}
+
+function  CheckIfOnlySecureTLSVersionConfigured {
+    param (
+        [String] $TLSVersion 
+    )
+    #Check if TLS Versions less than minimum required TLS versions are supported
+    $supportedTLSVersions = @();
+    $nonCompliantTLSVersions = @();
+    try {
+        $supportedTLSVersions = $TLSVersion.ToLower() -replace [Constants]::TLSversionPattern, '' -replace '_', '.';
```

Comment:
we are getting a single TLS value as input, then why are we converting into array, also $supportedTLSVersions is not appropriate name here.

Reply:
removed array and made checking only 1 variable.